examples/monaco-graphql-react-vite/src/App.tsx

```diff
@@ -156,52 +148,50 @@ export default function App() {
    * Handle the initial schema load
    */
   useEffect(() => {
-    if (!schema && !loading) {
-      setLoading(true);
-      void getSchema()
-        .then(data => {
-          if (!('data' in data)) {
-            throw new Error(
-              'this demo does not support subscriptions or http multipart yet',
-            );
-          }
-          initializeMode({
-            diagnosticSettings: {
-              validateVariablesJSON: {
-                [Uri.file('operation.graphql').toString()]: [
-                  Uri.file('variables.json').toString(),
-                ],
-              },
-              jsonDiagnosticSettings: {
-                validate: true,
-                schemaValidation: 'error',
-                // set these again, because we are entirely re-setting them here
-                allowComments: true,
-                trailingCommas: 'ignore',
-              },
-            },
-            schemas: [
-              {
-                introspectionJSON: data.data as unknown as IntrospectionQuery,
-                uri: 'myschema.graphql',
-              },
+    if (schema || loading) {
+      return;
+    }
+    setLoading(true);
+
+    void getSchema().then(data => {
+      const introspectionJSON =
+        'data' in data && (data.data as unknown as IntrospectionQuery);
+
+      if (!introspectionJSON) {
+        throw new Error(
+          'this demo does not support subscriptions or http multipart yet',
+        );
+      }
+      initializeMode({
+        diagnosticSettings: {
+          validateVariablesJSON: {
+            [Uri.file('operation.graphql').toString()]: [
+              Uri.file('variables.json').toString(),
             ],
-          });
+          },
+          jsonDiagnosticSettings: {
+            validate: true,
+            schemaValidation: 'error',
+            // set these again, because we are entirely re-setting them here
+            allowComments: true,
+            trailingCommas: 'ignore',
+          },
+        },
+        schemas: [{ introspectionJSON, uri: 'myschema.graphql' }],
+      });
 
-          setSchema(data.data);
-        })
-        .then(() => setLoading(false));
```

Comment:
here `.then(...)` is redundant, we can call `setLoading(false)` right after `setSchema(...)`

Reply:
true, it's not even proper async haha. maybe if we returned setSchema() but it's not necessary